lib/broadway_kafka/producer.ex

```diff
@@ -509,7 +516,29 @@ defmodule BroadwayKafka.Producer do
       |> Keyword.put(:processors, [updated_processor_entry | other_processors_entries])
       |> Keyword.put(:batchers, updated_batchers_entries)
 
-    {allocators, updated_opts}
+    {producer_mod, producer_opts} = opts[:producer][:module]
+
+    {extra_child_specs, initialized_client_config} =
+      if producer_opts[:shared_client] do
```

Comment:
Instead of calling `init` only if `:shared_client`, let's always call `init` here instead of calling `init` when the producer starts.